### ISSUE TITLE
Helper session sourcecontrol is wrong

### ISSUE THREAD

**Issue by aljesusg** [miq-consumption/manageiq-galaxy#31]
https://github.com/miq-consumption/manageiq-galaxy/blob/master/app/helpers/source_control_helper.rb#L84

@chargio Is this tested?
```ruby
[64, 73] in /Users/albertojesusgutierrezjuanes/Projects/manageiq-galaxy/app/controllers/v1/users/sessions_controller.rb
   64:       def verify_user!(code, request)
   65:         byebug
   66:         logger.debug 'Verifying that the user code is valid and authenticating user'
   67:         connection = source_control_server
   68:         access_token = code ? connection.exchange_code_for_token!(code) : nil # Verify github code and get token with it
=> 69:         github_user = connection.user
   70:         logger.debug "Valid code, finding or creating user #{github_user}"
   71:         user = User.first_or_create(github_user)
   72:         logger.debug 'Creating user session token'
   73:         token = Tiddle.create_and_return_token(user, request)
(byebug) access_token
{:access_token=>"b4131e02040c4861591715e967d04f55c1e28c6f",
 :token_type=>"bearer",
 :scope=>"user:email"}
(byebug) connection.user
nil

```

**Comment by aljesusg** [comment]
Where is the user" logging in github "

```ruby

   def exchange_code_for_token!(code)
      byebug
      github_token = @github_access.exchange_code_for_token(code)
      @github_access.access_token = github_token[:access_token] if github_token[:error] || github_token.nil?
      github_token
    end

    #
    # Returns the user
    # @return user [SourceControlUser]
    #
    def user
      @github_access.user if @github_access.user_authenticated?
    end

```

**Comment by aljesusg** [comment]
Something is wrong here setting the user
```ruby
(byebug) @github_access.exchange_code_for_token(code)
{:access_token=>"119d313904a7a9ad0e8d6e4ff1387f5b83fbf2a1",
 :token_type=>"bearer",
 :scope=>"user:email"}
(byebug) @github_access.user
*** Octokit::Unauthorized Exception: GET https://api.github.com/user?client_id=f49137725fc5870b0104&client_secret=(redacted): 401 - Requires authentication // See: https://developer.github.com/v3/users/#get-the-authenticated-user

nil
```

**Comment by chargio** [comment]
It is wrong.


```ruby
@github_access.access_token = github_token[:access_token] if github_token[:error] || 
```

Needed to be unless. I am not storing the code when it is real

**Comment by chargio** [comment]
Fixed by #32
